dask/array/core.py

```diff
@@ -964,12 +966,18 @@ def __init__(self, dask, name, chunks, dtype, shape=None):
             raise ValueError("You must specify the dtype of the array")
         self.dtype = np.dtype(dtype)
 
-    @property
-    def _args(self):
+        for plugin in _globals.get('array_plugins', ()):
+            result = plugin(self)
+            if result is not None:
+                self = result
+
+        return self
+
+    def __getnewargs__(self):
         return (self.dask, self.name, self.chunks, self.dtype)
 
     def __getstate__(self):
-        return self._args
+        return self.__getnewargs__()
 
     def __setstate__(self, state):
         self.dask, self.name, self._chunks, self.dtype = state
```

Comment:
Ah, it's because `Array` has `__slots__`, which requires `__getstate__` for pickle protocols < 2. If you specify protocol 2 (the max in python 2) then everything works. For protocol < 2, `__getnewargs__` won't be called at all. This means that unpickle won't apply the plugins on load. Protocols >= 2 however will apply both `__getnewargs__` and `__getsetstate__` if both are defined. This means that the state will be set twice on load (which may not be a big deal).

The best way may be to skip all of this, and just use `__reduce__` directly. This works in all protocols, ensures `__new__` is called on load, and sets the state only once.

```python
def __reduce__(self):
    return (Array, (self.dask, self.name, self.chunks, self.dtype))
```